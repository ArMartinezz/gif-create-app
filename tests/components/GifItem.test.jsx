import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("Pruebas para GifItem.jsx", () => {
  const title = "Dog GIF by ViralHog";
  const url =
    "https://media3.giphy.com/media/HzkZeQM6aHcodITVXf/giphy-downsized-medium.gif?cid=a1421da0rdcgni9gzmnijg3t8awj5jdf4metzs7r4yfftz6u&ep=v1_gifs_search&rid=giphy-downsized-medium.gif&ct=g";

  test("Match Snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("Mostrar imagen y contener alt text", () => {
    render(<GifItem title={title} url={url} />);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Debe mostrar el tÃ­tulo del componente", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
