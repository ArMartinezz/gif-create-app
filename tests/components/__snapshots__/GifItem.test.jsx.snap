// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para GifItem.jsx Match Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dog GIF by ViralHog"
      src="https://media3.giphy.com/media/HzkZeQM6aHcodITVXf/giphy-downsized-medium.gif?cid=a1421da0rdcgni9gzmnijg3t8awj5jdf4metzs7r4yfftz6u&ep=v1_gifs_search&rid=giphy-downsized-medium.gif&ct=g"
    />
    <p>
      Dog GIF by ViralHog
    </p>
  </div>
</div>
`;
